cmake/kokkos_functions.cmake

```diff
@@ -764,7 +764,7 @@ FUNCTION(kokkos_link_tpl TARGET)
 ENDFUNCTION()
 
 FUNCTION(COMPILER_SPECIFIC_OPTIONS_HELPER)
-  SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang GNU)
+  SET(COMPILERS NVIDIA PGI XL DEFAULT Cray Intel Clang AppleClang GNU HIP)
```

Comment:
Why would you list HIP here?

Reply:
Otherwise, `HIP` is just ignored when calling `COMPILER_SPECIFIC_FLAGS`, e.g., for setting warning flags.

Reply:
I don't see that we register any.  Did I miss something?

Reply:
Maybe, my choice of words was not the best. I was only talking about adding to `HIP` to the list of compilers handled here. We have https://github.com/kokkos/kokkos/blob/82e27afc4d51a6b7917650173adb0163939071a5/cmake/kokkos_functions.cmake#L778-L791
in this function. Thus, we don't set any flag if the compiler id doesn't match anything in the list.